packages/compose/src/hooks/use-throttle/index.js

```diff
@@ -15,12 +15,21 @@ import { useEffect } from '@wordpress/element';
  * including the function to throttle, so please wrap functions created on
  * render in components in `useCallback`.
  *
- * @param {...any} args Arguments passed to Lodash's `throttle`.
+ * @see https://docs-lodash.com/v4/throttle/
  *
- * @return {Function} Throttled function.
+ * @template {(...args: any[]) => void} TFunc TFunc
+ *
+ * @param {TFunc} fn The function to throttle.
+ * @param {number} [wait] The number of milliseconds to throttle invocations to.
+ * @param {import('lodash').ThrottleSettings} [options] The options object. See linked documentation for details.
+ * @return {TFunc & import('lodash').Cancelable} Throttled function.
```

Comment:
Should this use `DebouncedFunc` instead?

Reply:
Agreed, it seems like this could be a more up-to-date of typing it:

```suggestion
 * @return {import('lodash').DebouncedFunc<TFunc>} Throttled function.
```

Although it seems like there isn't any `DebouncedFunc` export, at least in the version of `lodash` that is being used

Reply:
Yeah I don't see `DebouncedFunc` in the declarations available to us 🤔

Reply:
Fine to leave it as-is then 👍